website/docs/en/plugins/rspack/html-rspack-plugin.mdx

```diff
@@ -289,6 +319,199 @@ module.exports = {
 };
 ```
 
+#### Use Template String
+
+Specify the HTML template content through `templateContent`:
+
+```js title="rspack.config.js"
+const rspack = require('@rspack/core');
+
+module.exports = {
+  plugins: [
+    new rspack.HtmlRspackPlugin({
+      title: "My HTML Template"
+      templateContent: `
+        <!DOCTYPE html>
+        <html>
+          <head>
+            <title><%= htmlRspackPlugin.options.title %></title>
+          </head>
+          <body></body>
+        </html>
+      `,
+    }),
+  ],
+};
+```
+
+#### Use Template Function
+
+Use a function to generate the HTML template content:
+
+- Pass the function in `templateContent`
+
+```js title="rspack.config.js"
+const rspack = require('@rspack/core');
+
+module.exports = {
+  plugins: [
+    new rspack.HtmlRspackPlugin({
+      title: "My HTML Template"
+      templateContent: ({ htmlRspackPlugin }) => `
+        <!DOCTYPE html>
+        <html>
+          <head>
+            <title>${htmlRspackPlugin.options.title}</title>
+          </head>
+          <body></body>
+        </html>
+      `,
+    }),
+  ],
+};
+```
+
+- Or pass a file path ending with `.js` or `.cjs` in `template`
```

Comment:
can we support mjs in the future

Reply:
maybe in the future, currently just require the file.